crates/bevy_render/src/render_graph/nodes/texture_node.rs

```diff
@@ -25,17 +27,57 @@ impl TextureNode {
             texture_descriptor,
             sampler_descriptor,
             handle,
+            has_changed: true,
         }
     }
 }
 
+impl TextureNode {
+    pub fn texture_descriptor(&self) -> &TextureDescriptor {
+        &self.texture_descriptor
+    }
+
+    pub fn texture_descriptor_mut(&mut self) -> &mut TextureDescriptor {
+        self.set_changed();
+        &mut self.texture_descriptor
+    }
+
+    pub fn sampler_descriptor(&self) -> &Option<SamplerDescriptor> {
+        &self.sampler_descriptor
+    }
+
+    pub fn sampler_descriptor_mut(&mut self) -> &mut Option<SamplerDescriptor> {
+        self.set_changed();
+        &mut self.sampler_descriptor
+    }
```

Comment:
I think it might be a little more idiomatic to return an `Option<&T>` and `Option<&mut T>`.
So I'd recommend changing these to:
```rust
    pub fn sampler_descriptor(&self) -> Option<&SamplerDescriptor> {
        self.sample_descriptor.as_ref()
    }

    pub fn sampler_descriptor_mut(&mut self) -> Option<&mut SamplerDescriptor> {
        self.set_changed();
        self.sample_descriptor.as_ref_mut()
    }
```

Reply:
I disagree, because `.as_ref()` creates a new `Option`.

library\core\src\option.rs:272
```rust
    #[inline]
    #[rustc_const_stable(feature = "const_option", since = "1.48.0")]
    #[stable(feature = "rust1", since = "1.0.0")]
    pub const fn as_ref(&self) -> Option<&T> {
        match *self {
            Some(ref x) => Some(x),
            None => None,
        }
    }
```

In this case we already have an option, so we can return a reference to that and let the user decide whether they need a reference to the contents or want to call some other function on it like `.map()` or `.is_some()`, without needing to make a new `Option`.

You'll see the same pattern in std, where a function will maybe return a reference, you get `Option<&_>` and where it returns a reference to something that may be there, it returns `&Option<_>`.

Reply:
Hmm I disagree with your arguments.

> `.as_ref()` creates a new `Option`.

While it creates a _new_ `Option`, it does not create a new `T`. This function is very cheep so performance isn't a worry here.

> In this case we already have an option, so we can return a reference to that and let the user decide whether they need a reference to the contents

I would argue that in every use case, you don't care about the `Option` part of it, you only care about the `T` that the `Option` is holding. Which is why having an `Option<T&>` is more intuitive. 
`&Option<T>` forces you to have a reference to an option,
`Option<&T>` only requires a reference to a T.

> You'll see the same pattern in std, where a function will maybe return a reference, you get Option<&_> and where it returns a reference to something that may be there, it returns &Option<_>

Um, I cannot find/think of any examples in the standard library that return a `&Option<T>` (but I could be wrong). And if I want "a reference to something that may be there", I'd expect an `Option<&T>`.

Reply:
Though we can leave the ultimatum to @cart haha :)

Reply:
I did look it up, but it's not particularly conclusive because both don't occur that often. I agree that a new Option isn't expensive and might be optimized out anyway and for some cases it is more convenient. Oh well...